content/guides/vercel-sdk-migration.md

```diff
@@ -0,0 +1,233 @@
+
+#### Simple Queries
+
+```typescript
+// Before (Vercel Postgres SDK)
+const { rows } = await sql`SELECT * FROM users WHERE id = ${userId}`;
+
+// After (Neon HTTP)
+const rows = await sql`SELECT * FROM users WHERE id = ${userId}`;
+
+// After (Neon WebSockets)
+const { rows } = await pool.query('SELECT * FROM users WHERE id = $1', [userId]);
+```
+
+#### Transactions
+
+```typescript
+// Before (Vercel Postgres SDK)
+import { db } from '@vercel/postgres';
+
+async function transferFunds(fromId: number, toId: number, amount: number) {
+  const client = await db.connect();
+  try {
+    await client.query('BEGIN');
+    await client.query('UPDATE accounts SET balance = balance - $1 WHERE id = $2', [
+      amount,
+      fromId,
+    ]);
+    await client.query('UPDATE accounts SET balance = balance + $1 WHERE id = $2', [amount, toId]);
+    await client.query('COMMIT');
+  } catch (e) {
+    await client.query('ROLLBACK');
+    throw e;
+  } finally {
+    client.release();
+  }
+}
+
+// After (Neon serverless driver - WebSockets)
+import { Pool } from '@neondatabase/serverless';
+
+async function transferFunds(fromId: number, toId: number, amount: number) {
+  const pool = new Pool({ connectionString: process.env.DATABASE_URL });
+  try {
+    await pool.query('BEGIN');
+    await pool.query('UPDATE accounts SET balance = balance - $1 WHERE id = $2', [amount, fromId]);
+    await pool.query('UPDATE accounts SET balance = balance + $1 WHERE id = $2', [amount, toId]);
+    await pool.query('COMMIT');
+  } catch (e) {
+    await pool.query('ROLLBACK');
+    throw e;
+  } finally {
+    await pool.end();
+  }
+}
+```
+
+### 4. Environment Variables
+
+Update your environment variable names if necessary:
+
+```bash
+# Before
+POSTGRES_URL="postgres://..."
+
+# After
+DATABASE_URL="postgres://..."
+```
+
+## Best practices
+
+1.  **Choose the right connection method**:
+
+    - Use HTTP (`neon()`) for single queries and simple transactions.
+    - Use WebSockets (`Pool`) for complex transactions and session-based operations.
+
+2.  **Connection management**:
+
+    - For HTTP queries, reuse the `sql` query function.
+    - For WebSocket connections in serverless environments, always close connections:
+
+    ```typescript
+    const pool = new Pool({ connectionString: process.env.DATABASE_URL });
+    try {
+      // Your queries here
+    } finally {
+      await pool.end();
+    }
+    ```
+
+3.  **Error Handling**:
+    ```typescript
+    try {
+      const result = await sql`SELECT * FROM users`;
+      return result;
+    } catch (error) {
+      console.error('Database error:', error);
+      throw new Error('Failed to fetch users');
+    }
+    ```
+
+## Working with ORMs
+
+If you're using an ORM, here are the recommended configurations:
+
+### Prisma
+
+Install the Prisma adapter for Neon, along with the Neon serverless driver and ws packages:
+
+```bash
+npm install @prisma/adapter-neon @neondatabase/serverless ws
```

Comment:
can we link our guides via cards instead? this will allow us to save the overhead in updating at multiple places.

Reply:
done